src/Transformer/DetermineHeroImages.php

```diff
@@ -86,81 +71,88 @@ final class DetermineHeroImages implements Transformer {
 	public function transform( Document $document, ErrorCollection $errors ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
 		$hero_image_elements = [];
 
-		foreach ( [ 'custom_header', 'custom_logo', 'featured_image', 'initial_image_block', 'initial_cover_block' ] as $hero_image_source ) {
+		foreach ( [ 'header_images', 'featured_image', 'initial_content_image' ] as $hero_image_source ) {
 			if ( count( $hero_image_elements ) < PreloadHeroImage::DATA_HERO_MAX ) {
 				$candidate = null;
 
 				switch ( $hero_image_source ) {
-					case 'custom_header':
-						$candidate = $this->get_custom_header( $document );
-						break;
-					case 'custom_logo':
-						$candidate = $this->get_custom_logo( $document );
+					case 'header_images':
+						$candidate = $this->get_header_images( $document );
 						break;
 					case 'featured_image':
 						$candidate = $this->get_featured_image( $document );
 						break;
```

Comment:
There's an issue here with the featured image. Namely, if there is a template with multiple posts on it, and the first post lacks a featured image but the second post has one, then this featured image will get prerendered:

> ![image](https://user-images.githubusercontent.com/134745/114319038-993b4d00-9ac4-11eb-9b27-fa29766d2b0b.png)

This issue is mitigated by preserving the `loading=lazy` on the original `img` as done in https://github.com/ampproject/amp-toolbox-php/pull/149 but it is not ideal. Consider this worse case where there is a content image in the first post, but a featured image in the second. The featured image from the second post will win out over the content image in the first:

> ![image](https://user-images.githubusercontent.com/134745/114319333-db18c300-9ac5-11eb-9759-74e0aad3beb9.png)

Reply:
See also an [e-commerce example](https://www.zotezo.com/in/reviews/greenworth-pure-aloe-vera-gel-200gm/?amp) (apparently using Astra and WooCommerce) in which the hero image is not marked up as a featured image, nor is it inside of `.entry-content`. Rather, it has a selector `main > div.ast-woocommerce-container > div.product-section > div > div > div:nth-child(1) > div > amp-img`.

This makes me think that we should perhaps eliminate the featured image check, in favor of getting the first images from the `main` element as hero candidates, or if no `main` then the first `.hentry`.